File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n+1];
        return calculateStep(n, dp);
    }
    public int calculateStep(int n, int[] dp){
        if(n<=2) return n;
        if(dp[n]!=0) return dp[n];
        dp[n] = calculateStep(n-1, dp) + calculateStep(n-2, dp);
        return dp[n];
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    # Top down + Memoization Approach
    def climbStairs(self, n: int) -> int:
        dp=[0]*(n+1)
        return self.calculateSteps(dp,n)
    def calculateSteps(self, dp, n):
        if n<=2:
            return n
        if dp[n]!=0:
            return dp[n]
        dp[n] = self.calculateSteps(dp,n-1)+self.calculateSteps(dp,n-2)
        return dp[n]